src/embed.js

```diff
@@ -25,13 +25,19 @@ var PREPROCESSOR = {
   'vega-lite': function(vljson) { return vl.compile(vljson).spec; }
 };
 
-function load(url, arg, el, callback) {
+function load(url, arg, prop, el, callback) {
   vg.util.load({url: url}, function(err, data) {
+    var opt = {};
     if (err || !data) {
       console.error(err || ('No data found at ' + url));
     } else {
       // marshal embedding spec and restart
-      var opt = !arg ? JSON.parse(data) : vg.util.extend({source: data}, arg);
+      if (!arg) { // Loading embed spec from URL
+        opt = JSON.parse(data);
+      } else {  // Loading vg/vl spec or config from URL
+        opt[prop] = JSON.parse(data);
+        opt = vg.util.extend({}, arg, opt);
```

Comment:
By always parsing the JSON here in the else branch, we discard the original source code formatting. The prior version of this method preserves the formatting. Should we drop this feature? (Also: if we decide to continue to preserve formatting, the prop argument should be changed from `'spec'` to `'source'`.)